pkg/referrerstore/oras/oras.go

```diff
@@ -117,30 +126,33 @@ func (store *orasStore) GetConfig() *config.StoreConfig {
 	return &store.rawConfig
 }
 
-func (store *orasStore) ListReferrers(ctx context.Context, subjectReference common.Reference, artifactTypes []string, nextToken string) (referrerstore.ListReferrersResult, error) {
-	// TODO: handle nextToken
-	registryClient, err := store.createRegistryClient(ctx, subjectReference)
+func (store *orasStore) ListReferrers(ctx context.Context, subjectReference common.Reference, artifactTypes []string, nextToken string, subjectDesc ...*ocispecs.SubjectDescriptor) (referrerstore.ListReferrersResult, error) {
+	repository, err := store.createRepository(ctx, subjectReference)
 	if err != nil {
 		return referrerstore.ListReferrersResult{}, err
 	}
 
-	ref := fmt.Sprintf("%s@%s", subjectReference.Path, subjectReference.Digest)
-	var referrerDescriptors []artifactspec.Descriptor
-	if artifactTypes == nil {
-		artifactTypes = []string{""}
-	}
-	for _, artifactType := range artifactTypes {
-		_, res, err := oras.Discover(ctx, registryClient.Resolver, ref, artifactType)
+	// resolve subject descriptor if not provided
+	var resolvedSubjectDesc *ocispecs.SubjectDescriptor
+	if len(subjectDesc) > 0 {
+		resolvedSubjectDesc = subjectDesc[0]
+	} else {
+		resolvedSubjectDesc, err = store.GetSubjectDescriptor(ctx, subjectReference)
```

Comment:
Do we need to resolve the descriptor again as I see subject desc passed to this method?

Reply:
The `subjectDesc` parameter is intended to act like an optional parameter. Should we just add the subjectDesc parameter as a hard required parameter. So users of the API would have to call the GetSubjectDescriptor method first? I was hesistant to do this because I saw the interface operated at the reference level.

Reply:
switched to normal parameter